atc/versioned_resource_types_test.go

```diff
@@ -0,0 +1,66 @@
+package atc_test
+
+import (
+	"github.com/concourse/concourse/atc"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("VersionedResourceTypes", func() {
+	var types atc.VersionedResourceTypes
+
+	BeforeEach(func() {
+		types = atc.VersionedResourceTypes{
+			{
+				ResourceType: atc.ResourceType{
+					Name:   "some-type",
+					Type:   "some-base-type",
+					Source: atc.Source{"some": "source"},
+				},
+				Version: atc.Version{"some": "version"},
+			},
+			{
+				ResourceType: atc.ResourceType{
+					Name:   "nested-type",
+					Type:   "some-type",
+					Source: atc.Source{"nested": "source"},
+				},
+				Version: atc.Version{"nested": "version"},
+			},
+			{
+				ResourceType: atc.ResourceType{
+					Name:   "overriding-type",
+					Type:   "overriding-type",
+					Source: atc.Source{"overriding": "source"},
```

Comment:
what is the type that this type trying to "override"?

Reply:
It overrides it by specifying the same type as its own type. This effectively overrides `overriding-type` everywhere *else* in the pipeline, while using the real "base" `overriding-type` for fetching the type itself.

As a real world use case, imagine someone overriding `registry-image` with a newer `registry-image` - they have to use the old `registry-image` to fetch the new one.

Reply:
uhh, I see. Should the type be named/typed with "some-base-type" ? So it matches the test case with context "when the type overrides a base type"

Reply:
Doing that would change the meaning of the other types, since one of them uses `some-base-type` - if I override it here, the other type would then refer to the overridden one. I can rename it to `overridden-base-type` tho!

Reply:
(done)